inspirehep/modules/workflows/tasks/merging.py

```diff
@@ -26,19 +26,101 @@
 
 from invenio_pidstore.resolver import PersistentIdentifier
 
-from inspirehep.utils.record import get_source, get_recid
+from inspirehep.utils.record import get_source
+from inspire_json_merger.inspire_json_merger import inspire_json_merge
+
+from ..utils import (
+    with_debug_logging,
+    retrieve_root_json,
+    retrieve_head_json,
+    store_head_json,
+    store_root_json,
+)
 
-from ..utils import with_debug_logging, retrieve_root_json as _retrieve_root_json
 
 @with_debug_logging
-def retrieve_root_json(obj, eng):
+def get_root(obj):
     """Retrieve the root JSON.
 
     Retrieves the root JSON corresponding to the current matched record and the
     provided source.
     The root version is stored in `obj.extra_data['root_json']`.
     """
-    source = get_source(obj.metadata)
-    recid = get_recid(obj.metadata)
-    record_uuid = PersistentIdentifier.get('literature', recid).object_uuid
-    obj.extra_data['root_json'] = _retrieve_root_json(record_uuid, source)
+    source = get_source(obj.data)
+    control_number = (obj.data.get('control_number', ''))
+    record_uuid = PersistentIdentifier.get('lit', control_number).object_uuid
+    return retrieve_root_json(record_uuid, source)
+
+
+@with_debug_logging
+def get_uuid_from_matched_record(obj):
+    """Retrieve the UUID from the JSON record.
+
+    Retrieves the record UUID .
+    The uuid is stored in `obj.extra_data['head_uuid']`.
+    """
+    recid = obj.extra_data['record_matches']['records'][0]['source']['control_number']
+    record_uuid = PersistentIdentifier.get('lit', recid).object_uuid
+    obj.extra_data['head_uuid'] = str(record_uuid)
+
+
+@with_debug_logging
+def get_head(obj):
+    """Retrieve the head from the JSON record.
+
+    Retrieves the head JSON corresponding to the current matched record.
+    The head version is stored in `obj.extra_data['head_json']`.
+    It assumes that inside obj.extra_data there is head_uuid key populated
+    """
+    if not obj.extra_data.get('head_uuid', False):
+        raise Exception
+
+    return retrieve_head_json(obj.extra_data['head_uuid'])
+
+
+def merge_articles(obj, eng):
+    # prepare root, it puts inside the extra_data['root_json'] the relative root
```

Comment:
Not sure these comments are good here. These side-effects should be documented in the `merge_articles` docstring.